Python/compile.c

```diff
@@ -6173,22 +6173,52 @@ compiler_pattern_mapping(struct compiler *c, pattern_ty p, pattern_context *pc)
     if (INT_MAX < size - 1) {
         return compiler_error(c, "too many sub-patterns in mapping pattern");
     }
+
     // Collect all of the keys into a tuple for MATCH_KEYS and
     // COPY_DICT_WITHOUT_KEYS. They can either be dotted names or literals:
+
+    // will hold only Constant_kind keys
+    PyObject *seen = PySet_New(NULL);
+    if (seen == NULL) {
+        return 0;
+    }
+
```

Comment:
Maybe add a comment to the effect of:

```suggestion
    // NOTE: In this loop, goto error on failure. Otherwise we'll leak seen!
```